app/services/create_reservation.rb

```diff
@@ -39,22 +42,47 @@ def user_already_has_reservation?
     user_already_has_reservation
   end
 
-  def event_is_at_capacity?
-    event_is_at_capacity = event_reservations.count >= event_capacity
+  def event_date_capacity?
+    event_date_capacity =
+      event_reservations.count >= event_date[:capacity]
```

Comment:
The capacity needs to be compared against all of the reservations for the date.
With the way it is right now, the event_reservations.count will be the number of reservations for the event slot not the event date (if the event slot id exists).